### PR TITLE
net processing: #26140 follow-ups

### PR DESCRIPTION
**Pull request by dergoegge**
Addresses left over feedback from #26140.

* https://github.com/bitcoin/bitcoin/pull/26140#discussion_r1153498543
* https://github.com/bitcoin/bitcoin/pull/26140#discussion_r1153499627

`mapRelay` is only accessed from the message processing thread and does not need to be kept in sync with anything validation specific, it is therfore perfectly fine to have it guarded by `g_msgproc_mutex`.

### CODE REVIEW

**Comment by dergoegge** [comment]
CI failure is intermittent, see #27380

**Comment by jnewbery** [comment]
utACK 3fa4c54ac54b2d738e0c43b57b5c232ee02fe3b3

**Comment by hebasto** [comment]
Concept ACK.

File: src/net_processing.cpp [hunk 1]
```diff
@@ -317,10 +317,6 @@ struct Peer {
     {
         return WITH_LOCK(m_tx_relay_mutex, return m_tx_relay.get());
     };
-    const TxRelay* GetTxRelay() const EXCLUSIVE_LOCKS_REQUIRED(!m_tx_relay_mutex)
-    {
-        return WITH_LOCK(m_tx_relay_mutex, return m_tx_relay.get());
-    };
```
Comment (hebasto) [inline src/net_processing.cpp:323]: Nice!

**Review by hebasto — APPROVED** [review]
ACK 3fa4c54ac54b2d738e0c43b57b5c232ee02fe3b3, I have reviewed the code and it looks OK, I agree it can be merged.

**Pull request merged**